src/app/api/utils/auth.ts

```diff
@@ -92,8 +92,26 @@ export const authOptions: AuthOptions = {
   ],
   callbacks: {
     // Unused arguments also listed to show what's available:
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     async jwt({ token, account, profile, trigger }) {
+      if (trigger === "update") {
+        // Update the FxA profile without requiring the user to log out and back in again.
+        if (!process.env.OAUTH_PROFILE_URI) {
+          throw new Error("env var not set OAUTH_PROFILE_URI");
+        }
+
+        const access_token = token.subscriber?.fxa_access_token;
+        if (!access_token) {
+          throw new Error("no access token");
+        }
+
+        const result = await fetch(process.env.OAUTH_PROFILE_URI, {
+          headers: {
+            Authorization: `Bearer ${access_token}`,
+          },
+        });
```

Comment:
@Vinnl  I noticed there's a `userInfo` property on our provider that does this same fetch basically, I can't find much info about what it's for in the docs though.

Reply:
Is there a way to call that, or does it serve a totally different purpose? Maybe both should call a utility function? not sure.

Reply:
OK I ended up refactoring this into a common function, it was less work than trying to build something compatible with context (all it really needs is an access token)